File: dart/isbn_verifier.dart
bool isValid(String isbn) {
  isbn = isbn.replaceAll('-', '');
  if (isbn.length != 10) {
    return false;
  }

  int n = 10;
  int sum = 0;
  for (int i = 0; i < isbn.length; i++) {
    if (isbn[i] == 'X') {
      sum += n * 10;
      continue;
    }

    int? d = int.tryParse(isbn[i]);
    if (d == null) {
      return false;
    }
    sum += n * d;
    n--;
  }

  return sum % 11 == 0;
}

File: go/isbn_verifier.go
package isbn

import (
	"strings"
	"unicode"
)

func IsValidISBN(isbn string) bool {
	isbn = strings.ReplaceAll(isbn, "-", "")

	if len(isbn) != 10 {
		return false
	}

	sum := 0
	for i, c := range isbn {
		if i == 9 && c == 'X' {
			sum += 10 * (10 - i)
		} else if unicode.IsDigit(c) {
			sum += int(c-'0') * (10 - i)
		} else {
			return false
		}
	}
	return sum%11 == 0
}
